File: clr-api/Models/PullRequest.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace clr_api.Models;

public class PullRequest
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string? Id { get; set; }

    [BsonElement("problemId")]
    public string ProblemUuid { get; set; }

    [BsonElement("upvoters")]
    public List<string> Upvoters { get; set; } = new();
    
    [BsonElement("body")]
    public string Body { get; set; }
    
    [BsonElement("author")]
    public string Author { get; set; }
    
}

File: clr-us/src/types/pullrequest.ts
export const isPullRequest = (pr: unknown): pr is PullRequest => {
  if (!pr || typeof pr !== 'object') {
    return false
  }
  const requiredKeys = ['id', 'problemUuid', 'upvoters', 'body', 'author']
  return requiredKeys.every((key) => Object.hasOwn(pr, key))
}

export type PullRequest = {
  id: string
  problemUuid: string
  upvoters: string[]
  body: string
  author: string
}
